zebra-network/src/peer/connection.rs

```diff
@@ -403,15 +404,12 @@ impl State {
                 span,
                 mut tx,
                 mut handler,
+                request_timer,
             } => {
                 // we have to get rid of the span reference so we can tamper with the state
                 let span = span.clone();
                 trace!(parent: &span, "awaiting response to client request");
-                let timer_ref = conn
-                    .request_timer
-                    .as_mut()
-                    .expect("timeout must be set while awaiting response");
-                let cancel = future::select(timer_ref, tx.cancellation());
+                let cancel = future::select(request_timer, tx.cancellation());
                 match future::select(peer_rx.next(), cancel)
                     .instrument(span.clone())
                     .await
```

Comment:
This match statement would be a lot easier to read if we mapped the `Either<Message, Either<Timeout, CancelChannel>>` to a custom enum with `Message`, `Timeout`, and `CancelChannel` variants.

But we should do that in a follow-up PR - it's not a critical change, because it isn't causing any panics.

Reply:
We also want a `select3` in `peer_set::initialize`, so we should do those changes together:
https://github.com/ZcashFoundation/zebra/blob/main/zebra-network/src/peer_set/initialize.rs#L310

This is now #1783.